### ISSUE TITLE
pylint gets class name wrong when constructing objects with a list of strings

### ISSUE THREAD

**Issue by vmalloc** [PyCQA/pylint#2009]
### Steps to reproduce

Run pylint on the following code:

```python
class Enum(object):
    def __init__(self, *values):
        self._values = {}
        for value in values:
            self._values[str(value).upper()] = value

    def __getattribute__(self, key):
        if key.startswith('_'):
            return super(Enum, self).__getattribute__(key)
        return self._values[str(key).upper()]


if __name__ == "__main__":
    Colors = Enum('red', 'green', 'blue')
    print(Colors.red)

```

### Current behavior

Pylint emits the following message:

```
************* Module testme
/tmp/testme.py:15: [E1101(no-member), ] Instance of 'red' has no 'red' member
```

### Expected behavior

This is actually related to #2008, but even if the no-member error *does* get thrown, one would expect it to say "Instance of 'Enum' has no 'red' member".

At the current state of things, it makes it really hard to add an exception to this check based on the class name...

### pylint --version output

```
pylint 1.8.3, 
astroid 1.6.2
Python 3.6.5 (default, Mar 30 2018, 06:41:53) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

**Comment by PCManticore** [comment]
I believe this might actually come from the enum inference tips we have in astroid: https://github.com/PyCQA/astroid/blob/master/astroid/brain/brain_namedtuple_enum.py#L235. Not sure how though, as we look explicitly for Enum related classes in the base classes of a class. Anyway, I might be wrong, regardless of where its coming from, it's definitely a bug.

**Comment by mbyrnepr2** [comment]
No `no-member` emitted on latest Pylint. Can this issue be closed?

**Comment by DanielNoord** [comment]
Thanks @mbyrnepr2
